dspace-api/src/test/java/org/dspace/uclouvain/itemEnhancer/UCLouvainItemEnhancerTest.java

```diff
@@ -600,4 +605,212 @@ public void testConsumerMultipleConfigurationsAndMultipleEnhancement() throws Ex
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void pollerTestWithSameTargetButTwoDifferentSources() throws Exception {
+        // Create two simple author-publication relations.
+        // The two persons are linked to the same publication via the name.
+        context.turnOffAuthorisationSystem();
+        Item person1 = ItemBuilder.createItem(context, collection)
+            .withEntityType("Person")
+            .withTitle("Pierre Kiroul")
+            .build();
+        Item person2 = ItemBuilder.createItem(context, collection)
+            .withEntityType("Person")
+            .withTitle("Jya Rivpa")
+            .build();
+        UUID person1Authority = person1.getID();
+        UUID person2Authority = person2.getID();
+
+        Item publication = ItemBuilder.createItem(context, collection)
+            .withTitle("Single publication")
+            .withEntityType("Publication")
+            .withAuthor("Pierre Kiroul", person1Authority.toString())
+            .withAuthor("Jya Rivpa", person2Authority.toString())
+            .build();
+        context.restoreAuthSystemState();
+
+        context.commit();
+        person1 = context.reloadEntity(person1);
+        person2 = context.reloadEntity(person2);
+        publication = context.reloadEntity(publication);
+
+        context.turnOffAuthorisationSystem();
+        itemService.setMetadataSingleValue(context, person1, "dc", "title", null, null, "Jean Gloutitou");
+        itemService.update(context, person1);
+        itemService.setMetadataSingleValue(context, person2, "dc", "title", null, null, "Jean Prendraideu");
+        itemService.update(context, person2);
+        context.restoreAuthSystemState();
+
+        context.commit();
+        person1 = context.reloadEntity(person1);
+        person2 = context.reloadEntity(person2);
+        publication = context.reloadEntity(publication);
+
+        // We now have one entry to process in the database.
+        List<ItemToEnhance> itemToEnhanceEntries = uclouvainItemEnhancerService.retrieveAllItemsToUpdate(context);
+        assertThat(itemToEnhanceEntries, hasSize(2));
+
+        // Let's trigger the poller to read the database and update the required items.
+        uclouvainItemEnhancerUpdatePoller.checkForItemUpdate();
+
+        // Reload the entities to have the latests modifications.
+        context.commit();
+        person1 = context.reloadEntity(person1);
+        person2 = context.reloadEntity(person2);
+        publication = context.reloadEntity(publication);
+
+        // Check that the publication was updated.
+        List<MetadataValue> currentAuthor = itemService.getMetadata(publication, "dc", "contributor", "author", null);
+        assertTrue(currentAuthor.stream().map(mv -> mv.getValue()).anyMatch(v -> v.equals("Jean Gloutitou")));
+        assertTrue(currentAuthor.stream().map(mv -> mv.getValue()).anyMatch(v -> v.equals("Jean Prendraideu")));
```

Comment:
COSMETIC: Test seems good; to be perfect, we should check the place of each author: ensure "Gloutitou" is the first and "Prendraideu" has the second place